### DISCUSSION TITLE
SSHFS is no longer maintained

### DISCUSSION THREAD

**Discussion by felinira** [ltsp/ltsp#729] [Ideas]
As [SSHFS](https://github.com/libfuse/sshfs#this-project-is-orphaned) is no longer being maintained there should probably be a different default way to mount user filesystems.

NFS over an SSH port forwarding tunnel could be a solution here to easily secure the NFS connection.

**Comment by alkisg** [comment]
The latest sshfs upload to debian unstable was [on 2022-03-22](https://tracker.debian.org/pkg/sshfs-fuse).
The latest sshfs release on github was [on 2022-05-26](https://github.com/libfuse/sshfs/commit/c91eb9a9a992f1a36c49a8e6f1146e45b5e1c8e7).

I mean, I'm not sure that sshfs is going away, it may just need some time to get a new maintainer. I don't think we should sound the alarm from the ltsp side just yet.

The NFS over SSH port forwarding idea is nice as an alternative; patches welcome! I think it will need some server side changes though, to make all the NFS ports static, as some of them are random.

Other ways for securing NFS traffic are NFSv4 with kerberos or even NFSv3 with wireguard. These too will need server-side changes.

**Comment by alkisg** [comment]
I'll convert this issue into a discussion; if sshfs doesn't find a new maintainer and it starts getting removed from distributions, we can increase the severity from "discussion" to "issue" then, thanks!

**Comment by lpadumorvan** [comment]
It's true that sshfs has a bug concerning the refreshing of ltsp client desktops (with chroot). You have to press F5 to refresh.
With us, the homes from the ltsp server are exported via NFS to the cups remote print server.
For PDF printing, CUPS prints the PDF directly on the desktop of the connected home.
But the desktop on the ltsp client does not refresh to see the created pdf file appear.
Maybe we should use SSH over NFS for our ltsp chroot clients.
DV
